File: VB/16_Print/MultiPage.vb
Imports Spire.Pdf
Imports Spire.Pdf.Print

Namespace MultiPage
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new instance of PdfDocument object
			Dim doc As New PdfDocument()

			' Load the PDF document from the specified file path
			doc.LoadFromFile("..\..\..\..\..\..\Data\MultiPage.pdf")

			' Configure the print settings to use a multi-page layout with 2 rows and 2 columns,
			' without rotating pages, in horizontal order
			doc.PrintSettings.SelectMultiPageLayout(2, 2, False, PdfMultiPageOrder.Horizontal)

			' Select a page range from page 3 to page 15 for printing
			doc.PrintSettings.SelectPageRange(3, 15)

			' Set the paper margins for printing to 10 units on all sides
			doc.PrintSettings.SetPaperMargins(10, 10, 10, 10)

			' Set the orientation of the printing to portrait (not landscape)
			doc.PrintSettings.Landscape = False

			' Print the document using the configured settings
			doc.Print()

			' Close the document after printing
			doc.Close()
		End Sub
	End Class
End Namespace

File: CS/16_Print/MultiPage.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Graphics;
using Spire.Pdf.Print;

namespace MultiPage
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            //Create a pdf document
            PdfDocument doc = new PdfDocument();

            //Load a file
            doc.LoadFromFile(@"..\..\..\..\..\..\Data\MultiPage.pdf");

            //Select muti page to one paper layout
            doc.PrintSettings.SelectMultiPageLayout(2, 2, false, PdfMultiPageOrder.Horizontal);

            //Set print page range
            doc.PrintSettings.SelectPageRange(3, 15);

            //Set paper margins,measured in hundredths of an inch
            doc.PrintSettings.SetPaperMargins(10, 10, 10, 10);

            //Indicating whether the page is printed in landscape or portrait orientation.
            doc.PrintSettings.Landscape = false;

            //Print document
            doc.Print();

            //Close the document
            doc.Close();
        }
    }
}
